include/envoy/upstream/load_balancer_type.h

```diff
@@ -52,6 +52,12 @@ class LoadBalancerSubsetInfo {
    * @return bool whether routing to subsets should take locality weights into account.
    */
   virtual bool localityWeightAware() const PURE;
+
+  /*
+   * @return bool whether the locality weights should be scaled according to how many
+   * hosts were removed from the original host set.
```

Comment:
I kind of feel this language is not quite on the mark, since it's not literally multiplying by the number of removed hosts, it's scaling to compensate for the fraction removed.